cmd/scaffold.go

```diff
@@ -1925,14 +1930,6 @@ func WithLogLevel(level string) Option {
 	}
 }
 
-// WithDB takes precedence over WithDataDir and datadir will be set to empty if DB is set using this option
-func WithDB(db *badger.DB) Option {
```

Comment:
not being used anywhere

Reply:
I think this was added to support the consensus follower

Reply:
Could you explain it a bit more? I can't find anywhere using it. 

I deleted it, and tested localnet, it ran fine.

Reply:
if you use a consensus follower as a library, you may want to use the same database for the blocks/etc and you app. so there's an option to provide your own db instead of the scaffold always creating one

Reply:
I see. We need to discuss this, because it creates extra complexity, as we are transitioning to pebble DB.

Reply:
I think it makes sense to remove the `db` field from `*BaseConfig`.

We have `NodeConfig` (defined in cmd/node_builder.go) that derives from the `BaseConfig`.

```
type NodeConfig struct {
  BaseConfig
  ...
}
```

NodeConfig contains modules that are initialized from the BaseConfig, such as database. 
Since `NodeConfig` already have a DB field, it doesn't make sense that `BaseConfig` also have a `DB` field. 

`BaseConfig` should only have `dbDir` field, and `NodeConfig` should only have initialized `DB` instance.

Reply:
I decided to rename this method into `WithBadgerDB` and keep the *badger.DB instance in both NodeConfig and BaseConfig, since the follower engine can only update the BaseConfig, eventually, BaseConfig.badgerDB will be assigned to NodeConfig.badgerDB. Same for `pebbleDB` field.